### PR TITLE
Teamcity

### PR DESCRIPTION
**Pull request by EWhite613**
#patch#

# Changelog
* Add Teamcity/Bitbucket comment on PR integration

### CODE REVIEW

**Comment by ember-cli-visual-acceptance** [comment]
PR ember-cli-visual-acceptance Report 
 [Imgur Album](http://imgur.com/a/dI9Na)

File: index.js [hunk 1]
```diff
@@ -228,7 +228,61 @@ function buildReport (params) {
...
+    }
+    var filename = options.project + '-' + options.repo + '-' + prNumber + '.png'
+    uploadToExpress(options.expressUrl, image, filename)
+    var githubApiPostOptions = {
+      'headers': {
+        'user-agent': 'visual-acceptance',
+        'Authorization': 'Basic ' + new Buffer(options.user + ':' + options.password, 'ascii').toString('base64')
+      },
+      'json': {
+        'text': '![PR ember-cli-visual-acceptance Report](' + 'http://frost.ciena.com:3000/' + filename + ')'
```
Comment (sglanzer) [inline index.js]: Seems odd to have something like this in the public repo

File: index.js [hunk 2]
```diff
@@ -228,7 +228,61 @@ function buildReport (params) {
     return response
   })
 }
+function buildTeamcityBitbucketReport (params, options, prNumber) {
+  return runCommand('phantomjs', ['vendor/html-to-image.js', 'visual-acceptance-report/report.html']).then(function (params) {
+    console.log('Sending to github')
```
Comment (sglanzer) [inline index.js]: Missed log

File: index.js [hunk 3]
```diff
@@ -228,7 +228,61 @@ function buildReport (params) {
...
+    var existingComment = false
+    for (var i = 0; i < bodyJSON.values.length; i++) {
+      if (bodyJSON.values[i].action === 'COMMENTED' && bodyJSON.values[i].comment.text.indexOf('![PR ember-cli-visual-acceptance Report]') > -1) {
+        existingComment = true
+        break
+      }
+    }
+    if (existingComment) {
+      response = {error: 'Comment already exists. Just updating image'}
+      console.log('Comment already exists. Just updating image')
```
Comment (sglanzer) [inline index.js]: Missed log

File: index.js [hunk 4]
```diff
@@ -504,6 +558,102 @@ module.exports = {
...
+        }, {
+          name: 'branch',
+          type: String,
+          default: 'master',
+          description: 'branch to push to'
+        }, {
+          name: 'user',
+          type: String,
+          default: '',
+          description: 'branch to push to'
```
Comment (sglanzer) [inline index.js]: Pretty sure all the descriptions after `branch` are mismatched ;)

File: index.js [hunk 5]
```diff
@@ -504,6 +558,102 @@ module.exports = {
...
+        }, {
+          name: 'repo',
+          type: String,
+          default: '',
+          description: 'branch to push to'
+        }, {
+          name: 'express-url',
+          type: String,
+          default: 'http://frost.ciena.com:3000/',
```
Comment (sglanzer) [inline index.js]: Same comment about this in a public repo

File: index.js [hunk 6]
```diff
@@ -504,6 +558,102 @@ module.exports = {
...
+          type: String,
+          default: '',
+          description: 'branch to push to'
+        }, {
+          name: 'repo',
+          type: String,
+          default: '',
+          description: 'branch to push to'
+        }, {
+          name: 'express-url',
```
Comment (sglanzer) [inline index.js]: Does this need to be an express server?  Or could it be named more generically?

**Comment by sglanzer** [comment]
Approved - but please remove the references to explicit servers

**Pull request merged**